go_modules/lib/dependabot/go_modules/file_fetcher.rb

```diff
@@ -37,23 +40,21 @@ def fetch_files
         ) do
           fetched_files = [go_mod]
           # Fetch the (optional) go.sum
-          fetched_files << go_sum if go_sum
+          fetched_files << T.must(go_sum) if go_sum
           fetched_files
         end
       end
 
       private
 
+      sig { returns(Dependabot::DependencyFile) }
       def go_mod
-        return @go_mod if defined?(@go_mod)
-
-        @go_mod = fetch_file_if_present("go.mod")
+        @go_mod ||= T.let(T.must(fetch_file_if_present("go.mod")), T.nilable(Dependabot::DependencyFile))
```

Comment:
@JamieMagee can you please confidence check me if this code is equivalent to what was before?

Reply:
The `if_present` suffix on the end of this makes is sounds like `nil` might actually be a valid value for `@go_mod`. So the `T.must` may be unnecessary here.

Reply:
Just checked, and the test failure is due to the `T.must` here.

Reply:
Thanks, I figured out this call tree by adding `nil` check to `fetch_files` method, tests are passing now